main.js

```diff
@@ -0,0 +1,249 @@
+      document.querySelectorAll('.works-modal .project-info-item'),
+    );
+
+    document.querySelector('.works-modal .project-title').innerHTML = modalData.project_title;
+    document.querySelector('.works-modal .project-info-title').innerHTML = modalData.project_info_title;
+    document.querySelector('.works-modal .Project-img').src = modalData.project_img;
+    document.querySelector('.works-modal .project-description').innerHTML = modalData.project_description;
+    document.querySelector('.works-modal .see-live').href = modalData.live_version;
+    document.querySelector('.works-modal .see-source').href = modalData.source_code;
+
+    modalCounterImg.forEach((infoImg) => {
+      infoImg.src = modalData.counter_image;
+      infoImg.alt = modalData.project_title;
+    });
+    infoItems.forEach((infoItem, idx) => {
+      infoItem.innerHTML = modalData.project_info_text[idx];
+    });
+    modalTags.forEach((tag, idx) => {
+      tag.innerHTML = modalData.tags[idx];
+    });
+
+    // Handle modal buttons
+    document.querySelector('.works-modal .see-live').onclick = () => {
+      window.location.href = modalData.live_version;
+    };
+    document.querySelector('.works-modal .see-source').onclick = () => {
+      window.location.href = modalData.source_code;
+    };
+  });
+});
+
+// Handle modal close button click
+closeModalBtn.addEventListener('click', () => {
+  modal.style.display = 'none';
+  body.style.overflow = 'auto';
+});
+
+// form validation
+const form = document.querySelector('.form-contact');
+const email = document.getElementById('email');
+const error = document.querySelector('small');
+form.addEventListener('submit', (e) => {
+  const message = [];
+
+  if (email.value !== email.value.toLowerCase()) {
+    message.push('Please use lowercase for email address');
+  }
+
+  if (message.length > 0) {
+    e.preventDefault();
+    error.innerHTML = message.join('<br/>');
+    email.style.border = '2px solid #df0000';
+    email.style.borderRadius = '3px';
+    email.style.backgroundColor = 'tomato';
+    error.style.backgroundColor = 'red';
+  }
+});
+
+// Local storage functionality
+let formData = {
+  full_name: '',
+  email: '',
+  message: '',
+};
+
+// retrieving the local storage
+if (localStorage.getItem('formData') !== null) {
+  const data = localStorage.getItem('formData');
+  formData = JSON.parse(data);
+}
+
+// getting form input fields and textarea
+const formElements = document.querySelectorAll('input, textarea');
+// looping through the elements
+formElements.forEach((element) => {
+  // displaying the formData values to input fields
+  element.value = formData[element.name];
+  // applying input event listener on elements
+  element.addEventListener('input', (e) => {
+    // setting the values of input fields to the respective keys in object
+    formData[e.target.name] = e.target.value;
+    // saving the data in local storage
+    localStorage.setItem('formData', JSON.stringify(formData));
```

Comment:
-  [ ] Great job. your local storage works as expected. kindly remove the comments for cleaner code practice. you can comment beginning of a section but not every single line of code. 😄